### ISSUE TITLE
CUDA_CHECK_ERROR on  code_objects/neurongroup_stateupdater_codeobject.cu:433 

### ISSUE THREAD

**Issue by akioolin** [brian-team/brian2cuda#152]
Hello All

I install brian2cuda on DGX machine. the cuda version is 8.0.
when I running the example which come from brian2cuda, there are some errors occurred.

the first one is add include header file in brian2cuda/templates/synapses_initialise_queue.cu
with "#include "unique.h" ". 

the second is the error in cuba_distdelays_cuda.py and . cuba_cuda.py
when I using "./run_all.sh cuda" in example directory, I found the following sample code could 
be run and no error. the following list shows no error examples.
STDP_multipost_standalone_cuda.py
STDP_neuronaltraces_standalone_cuda.py
STDP_standalone_cuda.py

in cuba_cuda.py, the error message is as following: 
ERROR: CUDA_CHECK_ERROR() failed at code_objects/neurongroup_stateupdater_codeobject.cu:433 : an illegal memory access was encountered
terminate called after throwing an instance of 'thrust::system::system_error'
  what():  cudaFree in free: driver shutting down

in cuba_distdelays_cuda.py, the error message is as following:
terminate called after throwing an instance of 'thrust::system::system_error'
  what():  an illegal memory access was encountered

Does any body know how to fix this bug?

Best Regards,
Akio

**Comment by moritzaugustin** [comment]
Hi Akio,
the brian2cuda implementation is currently under heavy final development and is not ready for production use yet and not prepared for installation by end users. But we are happy to announce that there will be a release 1.0 very soon (likely end of October) about which we will inform you also here.  
Best, Moritz

**Comment by akioolin** [comment]
Hello Moritz,

Thank you very much. 

Right now, we are trying to find a way to accelerate the SNN simulation approach by using brian2.
I'll wait this project and trace more code to see how to utilize this work to help us to make more progress on SNN research.

Best Regards,
Akio

**Comment by akioolin** [comment]
Hello Moritz,

after some trace about the brian2/device structure, is it the new brian2cuda is merged into brian2 in device directory, like cpp_standalone device, as a new cuda_standalone device?

Best Regards,
Akio

**Comment by moritzaugustin** [comment]
yes, you are right. changing from generating c++ code with `cpp_standalone` we are generating gpu code simply by inserting below `import brian2`the following lines:

```import brian2cuda
set_device('cuda_standalone')```

however, we will not merge brian2 and brian2cuda but release brian2cuda as separate package
